### PR TITLE
Scrape locations

### PR DESCRIPTION
**Pull request by piticent123**
It should correctly scrape the location list instead of hardcoding it. However someone should definitely look over the changes as there is a good chance I made a mistake

### CODE REVIEW

**Comment by piticent123** [comment]
fixes #30

File: app.js [hunk 1]
```diff
@@ -1,64 +1,63 @@
-var express = require('express');
-var app = express();
-var redis = require("redis");
-var redisOptions = {
-  host:'redis',
-  port:6379,
-  total_retry_time:300000
+const express = require('express');
+const app = express();
```
Comment (vidia) [inline app.js]: Yay!

File: app.js [hunk 2]
```diff
@@ -1,64 +1,63 @@
...
-app.get('/Laundry/v2/demo', require('./routes/Laundry/get_demo'))
-app.get('/Laundry/:location', require('./routes/Laundry/get_info'))
-app.get('/Laundry/location/all', require('./routes/Laundry/get_all'))
-app.get('/Laundry/v2/:location', require('./routes/Laundry/get_test')) //this should be changed from the 'v2' as it is NOT a v2
-
-app.get('/v1/location/all', require('./routes/Laundry/get_all'))
-app.get('/v1/location/:location', require('./routes/Laundry/get_info'))
+app.get('/Laundry/test', require('./routes/Laundry/get_test'));
+app.get('/Laundry/status', require('./routes/Laundry/get_status'));
+app.get('/Laundry/v2/demo', require('./routes/Laundry/get_demo'));
```
Comment (vidia) [inline app.js]: We should move this /version thing to a variable or a different router so we can update it globally.

File: app.js [hunk 3]
```diff
@@ -1,64 +1,63 @@
...
-app.get('/Laundry/test', require('./routes/Laundry/get_test'))
-app.get('/Laundry/status', require('./routes/Laundry/get_status'))
-app.get('/Laundry/v2/demo', require('./routes/Laundry/get_demo'))
-app.get('/Laundry/:location', require('./routes/Laundry/get_info'))
-app.get('/Laundry/location/all', require('./routes/Laundry/get_all'))
-app.get('/Laundry/v2/:location', require('./routes/Laundry/get_test')) //this should be changed from the 'v2' as it is NOT a v2
-
-app.get('/v1/location/all', require('./routes/Laundry/get_all'))
-app.get('/v1/location/:location', require('./routes/Laundry/get_info'))
+app.get('/Laundry/test', require('./routes/Laundry/get_test'));
```
Comment (vidia) [inline app.js]: What is this?

File: app.js [hunk 4]
```diff
@@ -1,64 +1,63 @@
...
+app.get('/Laundry/status', require('./routes/Laundry/get_status'));
```
Comment (vidia) [inline app.js]: This needs to be versioned.

File: app.js [hunk 5]
```diff
@@ -1,64 +1,63 @@
...
+app.get('/Laundry/v2/demo', require('./routes/Laundry/get_demo'));
+app.get('/Laundry/:location', require('./routes/Laundry/get_info'));
+app.get('/Laundry/location/all', require('./routes/Laundry/get_all'));
```
Comment (vidia) [inline app.js]: Oh wait. we need to delete some of these.... disregard most of my above comments.

File: refreshCache.js [hunk 6]
```diff
@@ -1,50 +1,45 @@
...
-      }
-    });
-  });
+function grabHTML(url) {
+	return new Promise(function (resolve, reject) {
+		request(url, function (err, response, body) {
+			let results = [];
+			if (!err && response.statusCode === 200) {
+				results = parseHTML(body);
+				resolve(results);
```
Comment (vidia) [inline refreshCache.js]: We might want to switch to tabs so we can each set a tab width. 

Yes, I am engaging the tabs vs spaces debate.

File: routes/Laundry/get_locations.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+const cheerio = require('cheerio');
+const request = require('request');
+
+module.exports = function getLocations() {
```
Comment (vidia) [inline routes/Laundry/get_locations.js]: Can we change this to "Get Location Urls" OR "Get location Ids" to differentiate it from /locations/all?

File: routes/Laundry/get_status.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-module.exports = function(req, res){
-	var status = ['Available', 'In Use', 'Almost Done', 'End of Cycle', 'Out of Order', 'Offline', 'Ready To Start'];
+module.exports = function (req, res) {
+	let status = ['Available', 'In Use', 'Almost Done', 'End of Cycle', 'Out of Order', 'Offline', 'Ready To Start'];
```
Comment (vidia) [inline routes/Laundry/get_status.js]: Not a fix you need to make, but I want to find a way to make these dynamic as well :/

File: routes/Laundry/get_url.js [hunk 9]
```diff
@@ -1,20 +1,9 @@
...
-	}
-	return locations[input];
-}
\ No newline at end of file
+let getLocations = require('./get_locations');
+
+module.exports = function (input) {
+	return new Promise(function (resolve, reject) {
+		getLocations().then(function(locations) {
+			resolve(locations.reduce((acc, item) => item.name === input ? item.url : ""));
```
Comment (vidia) [inline routes/Laundry/get_url.js]: Where is this cached? I am not seeing the code right now.

**Pull request merged**